src/fields/field_fe.cc

```diff
@@ -290,11 +310,132 @@ void FieldFE<spacedim, Value>::value_list (const std::vector< Point >  &point_li
 
 
 template <int spacedim, class Value>
+void FieldFE<spacedim, Value>::init_from_input(const Input::Record &rec, const struct FieldAlgoBaseInitData& init_data) {
+	this->init_unit_conversion_coefficient(rec, init_data);
+	flags_ = init_data.flags_;
+
+
+	// read mesh, create tree
+    {
+       reader_file_ = FilePath( rec.val<FilePath>("mesh_data_file") );
+       auto reader_data = ReaderInstance::get_instance(reader_file_);
+       source_mesh_ = reader_data.mesh_;
+       source_mesh_->get_bih_tree(); // only create BIH tree
+	   // no call to mesh->setup_topology, we need only elements, no connectivity
+    }
+
+	field_name_ = rec.val<std::string>("field_name");
+}
+
+
+
+template <int spacedim, class Value>
+void FieldFE<spacedim, Value>::set_mesh(const Mesh *mesh, bool boundary_domain) {
```

Comment:
I see. So, set_mesh is virtual so it should work even if I prefere to just set the mesh pointer in that method. Nevertheless, the base implementation of set_mesh is called in update_history. First make
FieldFE::set_mesh as override to be sure that argument types match. Second, output typename of the field object just before set_mesh is called to get which instance was in fact created by the factory. ... so debug FieldFE creation process.